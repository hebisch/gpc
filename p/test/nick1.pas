{ BUG: modules don't initialize their interface's variables }

program my_main (input,output);
import nick1m1;
       nick1m2;

Var
   another_str : mystr := '7654321';
   another_int : integer := 361;

begin
   sSet ('0123456');
   writeln ('1. my_main another_str:',another_str,'.');
   another_str := sGet;
   writeln ('2. my_main another_str:',another_str,'.');

   iSet (19);
   writeln ('1. my_main another_int:',another_int,'.');
   another_int := iGet;
   writeln ('2. my_main another_int:',another_int,'.');

end.
