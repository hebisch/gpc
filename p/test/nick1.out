Load nick1m1
nick1m2 sset str_arg:0123456.
nick1m2 sset my_str_var:0123456.
1. my_main another_str:7654321.
2. my_main another_str:0123456.
nick1m2 sset int_arg:19.
nick1m2 sset my_int_var:19.
1. my_main another_int:361.
2. my_main another_int:19.
